src/Build/BackEnd/Components/SdkResolution/SdkResolution.cs

```diff
@@ -0,0 +1,198 @@
+        private IList<SdkResolver> _resolvers;
+
+        /// <summary>
+        ///     Create an instance with a specified resolver assembly loading strategy. Used
+        ///     for testing purposes.
+        /// </summary>
+        /// <param name="sdkResolverLoadingStrategy">Resolver loading strategy.</param>
+        internal SdkResolution(SdkResolverLoadingStrategy sdkResolverLoadingStrategy)
+        {
+            _sdkResolverLoadingStrategy = sdkResolverLoadingStrategy;
+        }
+
+        internal static SdkResolution Instance { get; } = new SdkResolution(new SdkResolverLoadingStrategy());
+
+        /// <summary>
+        ///     Get path on disk to the referenced SDK.
+        /// </summary>
+        /// <param name="sdk">SDK referenced by the Project.</param>
+        /// <param name="logger">Logging service.</param>
+        /// <param name="buildEventContext">Build event context for logging.</param>
+        /// <param name="importElementLocation">Location of the element within the project which referenced the SDK.</param>
+        /// <param name="solutionPath">Path to the solution if known.</param>
+        /// <returns>Path to the root of the referenced SDK.</returns>
+        internal string GetSdkPath(ReferencedSdk sdk, ILoggingService logger, BuildEventContext buildEventContext,
+            ElementLocation importElementLocation, string solutionPath)
+        {
+            ErrorUtilities.VerifyThrowInternalNull(sdk, nameof(sdk));
+            ErrorUtilities.VerifyThrowInternalNull(logger, nameof(logger));
+            ErrorUtilities.VerifyThrowInternalNull(buildEventContext, nameof(buildEventContext));
+            ErrorUtilities.VerifyThrowInternalNull(importElementLocation, nameof(importElementLocation));
+
+            if (_resolvers == null) Initialize(logger, buildEventContext, importElementLocation);
+
+            var results = new List<SdkResultImpl>();
+
+            try
+            {
+                var buildEngineLogger = new SdkLoggerImpl(logger, buildEventContext);
+                foreach (var sdkResolver in _resolvers)
+                {
+                    var context = new SdkResolverContextImpl(buildEngineLogger, importElementLocation.File, solutionPath);
+                    var resultFactory = new SdkResultFactoryImpl(sdk);
+                    try
+                    {
+                        var result = (SdkResultImpl)sdkResolver.Resolve(sdk, context, resultFactory);
+                        if (result != null && result.Success)
+                        {
+                            LogWarnings(logger, buildEventContext, importElementLocation, result);
+                            return result.Path;
+                        }
+
+                        results.Add(result);
+                    }
+                    catch (Exception e)
+                    {
+                        logger.LogFatalBuildError(buildEventContext, e, new BuildEventFileInfo(importElementLocation));
```

Comment:
This `catch` looks redundant with the overall one, except it doesn't rethrow? Just to avoid onion-peeling on resolver exceptions?